src/render.js

```diff
@@ -1,9 +1,23 @@
 import { EMPTY_OBJ, EMPTY_ARR } from './constants';
-import { commitRoot } from './diff/index';
-import { diffChildren } from './diff/children';
+import { commitRoot, diff } from './diff/index';
 import { createElement, Fragment } from './create-element';
 import options from './options';
 
+function addRenderRoot(vnode, parent) {
```

Comment:
Should this live with Portal in compat? I think only Portals should add to the render root. Calling `render` multiple times on the same container is suppose to replace the elements in the container. Portal is special cuz it doesn't. Then perhaps you can do `root.children.push()`

Reply:
So instead of `render(wrap, container)` in Portal, perhaps it should be `addRenderRoot(wrap, container)` and `addRenderRoot` just calls `render` if there is no existing root, else it appends the child to the existing root's children and runs `forceUpdate`.

Reply:
In that case we need to be able to check if prev and nextNode are different no? Else we call addRenderRoot every render

Reply:
Well, something needs to change anyway too much tests goofing up, trying to find out why first :D